radix-engine/tests/proof/src/vault_proof.rs

```diff
@@ -0,0 +1,34 @@
+use scrypto::prelude::*;
+
+blueprint! {
+    struct VaultProof {
+        vault: Vault,
+    }
+
+    impl VaultProof {
+        pub fn new(bucket: Bucket) -> ComponentId {
+            Self {
+                vault: Vault::with_bucket(bucket),
+            }
+            .instantiate()
+        }
+
+        pub fn create_clone_drop_vault_proof(&self, amount: Decimal) {
+            let proof = self.vault.create_proof();
+            let clone = proof.clone();
+
+            assert_eq!(self.vault.amount(), amount);
+            assert_eq!(proof.amount(), amount);
+            assert_eq!(clone.amount(), amount);
+
+            clone.drop();
+            proof.drop();
+        }
+
+        pub fn use_vault_proof_for_auth(&self, to_burn: Bucket) {
+            self.vault.authorize(|| {
```

Comment:
We also have the resource system function authorize(). We should come to agreement on best interface for this.